lib/bulbs-elements/store/store.js

```diff
@@ -42,13 +65,15 @@ function dispatchFunction (store, action, payload) {
     console.time(`dispatch ${action.type}`);
   }
 
-  let nextState = cloneObject(store.state);
-  let actionState = nextState[action.fieldName];
-  let actionNextState = cloneObject(actionState);
-  actionNextState = action(actionNextState, payload, store);
-  nextState[action.fieldName] = actionNextState;
+  let priorState = store.state;
+  let nextState = Object.assign({}, priorState);
+  let priorFieldValue = priorState[action.fieldName];
+  let nextFieldValue = cloneObject(priorFieldValue);
```

Comment:
Yeah, this object copying here is two steps:

1. shallow copy entire store
1. deep copy the value of the field undergoing transition

This is good for reasons:

1. we don't incur the cost of copying the entire state for every action
2. react can optimize it's re-render check by comparing object identity.

    `store.state === store.state` simple check re-render components that have the full state as a prop

    `store.state.foo === store.state.foo`  simple check to re-render components that have a field as a prop

This is not perfect for reasons:

1. react doesn't get such a simple identity check for more deeply nested fields

But, bulbs-elements elements are expected to be small components/widgets with small trees of state, not large applications with huge forests of state. This should be kept in mind as we consider additions to the set of elements. So far I don't see anything we've come up with as going out of scope.